### PR TITLE
SCP-3953: Improve cost inference for verifyEcdsaSecp256k1Signature 

### PR DESCRIPTION
**Pull request by kwxm**
... and update the cost model.

As explained [here](https://github.com/input-output-hk/plutus/pull/4591#issuecomment-1120797931) there was some confusion regarding the costing for `verifyEcdsaSecp256k1Signature` because verification fails rapidly (returning `False`) for some apparently reasonable signatures. This improves the R code that infers the cost, and also adds some notes documenting the unexpected (or at least non-obvious) behaviour.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

**Comment by kwxm** [comment]
@michaelpj This should probably be included in the backport at #4600.

File: plutus-core/cost-model/data/models.R [hunk 1]
```diff
@@ -504,8 +504,25 @@ modelFun <- function(path) {
...
     }
 
-    verifyEcdsaSecp256k1SignatureModel <- constantModel ("VerifyEcdsaSecp256k1Signature")
-
+    ## The verifyEcdsaSecp256k1Signature function returns quickly for 50% of
+    ## randomly-generated signatures because the Bitcoin implementation that
+    ## underlies it enforces a requirement that the lower of the two possible
+    ## values of the s component of the signature is used, returning immediately
+    ## if that's not the case: see Note [ECDSA secp256k1 signature verification]
+    ## in Builtins.hs.  This gives us bencharking results where the times fall
```
Comment (bezirg) [inline plutus-core/cost-model/data/models.R:512]: typo: bencharking -> benchmarking

**Review by bezirg — APPROVED** [review]

File: plutus-core/cost-model/data/models.R (hunk 1 above)
Comment (kwxm) [reply plutus-core/cost-model/data/models.R:512]: Thank you.

**Comment by michaelpj** [comment]
Has conflicts

**Pull request merged**